code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java

```diff
@@ -92,6 +95,34 @@ public boolean isError() {
         return false;
     }
 
+    public boolean isHostActive() {
+        for (DeploymentUnitInstance deployUnitInstance : getDeploymentUnitInstances()) {
+            if (deployUnitInstance instanceof ExternalDeploymentUnitInstance) {
```

Comment:
The point is valid. But we shouldn't assume that the instance is the common case. Like today DNS service doesn't have corresponding UnitInstance, but if we decide to add it in the future - lets say some extra functionality required for this service besides configuring DNS - this implementation won't have an instance, and will break things.